test/Modules/UngradedModules.jl

```diff
@@ -0,0 +1,675 @@
+  N3 = Oscar.SubQuo(F3,R[x^2*y+13*x*y+2x-1-x*y^2 0 2*x*y-x*y^2; y^4-x*y^2 3*x-x^4 -1-x*y^2],R[2*y^2 2*x^3 2*y^2])
+  Q3,p3 = quo(M3,N3,:store)
+
+  @test iszero(quo(M3,M3))
+  @test iszero(Q3)
+  for k=1:5
+    elem = Oscar.SubQuoElem(Oscar.sparse_row(array_to_matrix([randpoly(R) for _=1:1,i=1:1])), M3)
+    @test p3(elem) == Oscar.map_canonically(Q3, elem)
+    @test iszero(p3(elem))
+  end
+end
+
+@testset "submodules" begin
+  R, (x,y) = PolynomialRing(QQ, ["x", "y"])
+
+  F2 = Oscar.FreeMod(R,2)
+  M1 = Oscar.SubQuo(F2,R[x^2*y+x*y x*y^2-x; x+x*y^2 y^3],R[x^2 y^3-x])
+  S1,i1 = Oscar.sub(M1, [M1(Oscar.sparse_row(R[1 1])),M1(Oscar.sparse_row(R[y -x]))], :store)
+
+  @test S1 == Oscar.SubQuo(F2,R[x*y^2+x^3-x^2 x*y^3-x*y-x^2; x^2*y+x*y^2+x*y-x^2+x x*y^2],R[x^2 y^3-x])
+  for k=1:5
+      elem = S1(Oscar.sparse_row(array_to_matrix([randpoly(R) for _=1:1,i=1:2])))
+      @test i1(elem) == Oscar.map_canonically(M1, elem)
+  end
+
+  M2 = Oscar.SubQuo(F2,R[x*y^2+x*y x^3+2*y; x^4 y^3; x^2*y^2+y^2 x*y],R[x^3-y^2 y^4-x-y])
+  S2,i2 = sub(M2,[M2(Oscar.sparse_row(R[x*y -x*y^2 x*y])),M2(Oscar.sparse_row(R[x 0 -1]))], :store)
+
+  @test S2 == Oscar.SubQuo(F2,R[x^2*y^3+x^2*y^2+x^3*y^3+x*y^3-x^5*y^2 x^4*y+2*x*y^2-x*y^5+x^2*y^2; x^2*y-y^2 x^4+x*y],R[x^3-y^2 y^4-x-y])
+  for k=1:5
+      elem = S2(Oscar.sparse_row(array_to_matrix([randpoly(R) for _=1:1,i=1:2])))
+      @test i2(elem) == Oscar.map_canonically(M2, elem)
+  end
+
+  M3 = Oscar.SubQuo(F2,R[x*y^2 x^3+2*y; x^4 y^3; x*y+y^2 x*y],R[x^3-y^2 y^4-x-y])
+  elems = [M3(Oscar.sparse_row(R[0 6 0])),M3(Oscar.sparse_row(R[9 0 -x])),M3(Oscar.sparse_row(R[0 0 -42]))]
+  S3,i3 = sub(M3,elems,:store)
+
+  @test S3 == M3
+  for k=1:5
+      elem = S3(Oscar.sparse_row(array_to_matrix([randpoly(R) for _=1:1, i=1:3])))
+      @test i3(elem) == Oscar.map_canonically(M3, elem)
+  end
+end
+
+@testset "Hom module" begin
+	R, (x0,x1,x2,x3,x4,x5) = PolynomialRing(QQ, ["x0", "x1", "x2", "x3", "x4", "x5"])
+	f1=R[-x2*x3 -x4*x5 0; x0*x1 0 -x4*x5; 0 x0*x1 -x2*x3]'
+	g1=R[x0*x1 x2*x3 x4*x5]'
+	M = cokernel(f1)
+	N = cokernel(g1)
+	SQ = Oscar.hom(M,N)[1]
+
+	#println("-------")
+	#display(SQ)
+	#println("--------")
+	#display(simplify(SQ)[1])
+	#println("--------")
+
+	f2 = R[0 0]
+	M = cokernel(f1)
+	N = cokernel(f2)
+	SQ = hom(M,N)[1]
+
+	#println("-------")
+	#display(SQ)
+	#println("--------")
+	#display(simplify(SQ)[1])
+	#println("--------")
+
+	function free_module_SQ(ring,n)
+		A = one(AbstractAlgebra.MatrixSpace(ring,n,n))
+		B = zero(AbstractAlgebra.MatrixSpace(ring,1,n))
+		return Oscar.SubQuo(A,B)
+	end
+
+	function free_module_SQ(F::Oscar.FreeMod)
+		return Oscar.SubQuo(F,gens(F))
+	end
+
+	# test if Hom(R^n,R^m) gives R^(m*n): (there is a dedicated function for free modules but this is a simple test for the function for subquos)
+	for n=1:5
+		for m=1:5
+			M=free_module_SQ(R,n)
+			N=free_module_SQ(R,m)
+			#SQ = Oscar.hom(M,N)[1] # it's weird that it fails without simplifying
```

Comment:
Is this a bug in your code here? Or in existing Oscar code? In the latter case, did you file a bug report?